packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/OutputLibrary.cs

```diff
@@ -10,58 +10,60 @@ namespace Microsoft.Generator.CSharp
         private IReadOnlyList<ClientProvider>? _clients;
 
         public OutputLibrary()
         {
-            EnumMappings = new Dictionary<InputEnumType, EnumProvider>();
-            ModelMappings = new Dictionary<InputModelType, ModelProvider>();
+            EnumMappings = new Dictionary<InputEnumType, TypeProvider>();
+            ModelMappings = new Dictionary<InputModelType, TypeProvider>();
 
             _allModels = new(InitializeAllModels);
         }
 
-        private readonly Lazy<(EnumProvider[] Enums, ModelProvider[] Models)> _allModels;
-
-        private (EnumProvider[] Enums, ModelProvider[] Models) InitializeAllModels()
+        private (TypeProvider[] Enums, TypeProvider[] Models) InitializeAllModels()
         {
             var input = CodeModelPlugin.Instance.InputLibrary.InputNamespace;
 
-            var enums = new EnumProvider[input.Enums.Count];
+            var enums = new TypeProvider[input.Enums.Count];
             for (int i = 0; i < enums.Length; i++)
             {
                 var inputEnum = input.Enums[i];
-                var enumType = EnumProvider.Create(inputEnum);
+                var cSharpEnum = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputEnum);
+                TypeProvider enumType = cSharpEnum.Implementation;
                 enums[i] = enumType;
                 EnumMappings.Add(inputEnum, enumType);
             }
 
-            var models = new ModelProvider[input.Models.Count];
+            var models = new TypeProvider[input.Models.Count];
             for (int i = 0; i < models.Length; i++)
             {
                 var inputModel = input.Models[i];
-                var model = new ModelProvider(inputModel);
-                models[i] = model;
-                ModelMappings.Add(inputModel, model);
+                var cSharpModel = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputModel);
+                TypeProvider modelType = cSharpModel.Implementation;
+                models[i] = modelType;
+                ModelMappings.Add(inputModel, modelType);
             }
 
             return (enums, models);
         }
 
-        public IReadOnlyList<EnumProvider> Enums => _enums ??= BuildEnums();
-        public IReadOnlyList<ModelProvider> Models => _models ??= BuildModels();
+        public IReadOnlyList<TypeProvider> Enums => _enums ??= BuildEnums();
+        public IReadOnlyList<TypeProvider> Models => _models ??= BuildModels();
         public IReadOnlyList<ClientProvider> Clients => _clients ??= BuildClients();
 
-        public IDictionary<InputEnumType, EnumProvider> EnumMappings { get; }
-        public IDictionary<InputModelType, ModelProvider> ModelMappings { get; }
+        private IDictionary<InputEnumType, TypeProvider> EnumMappings { get; }
+        private IDictionary<InputModelType, TypeProvider> ModelMappings { get; }
+
+        private readonly Lazy<(TypeProvider[] Enums, TypeProvider[] Models)> _allModels;
```

Comment:
I think this code was already like this, but why do we use `Lazy` here as opposed to just calling InitializeAllModels from the ctor? Won't we always need to get initialize the models when generating a library?